src/pages/PoolsPage/v3/SupplyLiquidityV3/index.tsx

```diff
@@ -0,0 +1,512 @@
+  );
+  const prevDerivedMintInfo = usePrevious({ ...derivedMintInfo });
+
+  const mintInfo = useMemo(() => {
+    if (
+      (!derivedMintInfo.pool ||
+        !derivedMintInfo.price ||
+        derivedMintInfo.noLiquidity) &&
+      prevDerivedMintInfo
+    ) {
+      return {
+        ...prevDerivedMintInfo,
+        pricesAtTicks: derivedMintInfo.pricesAtTicks,
+        ticks: derivedMintInfo.ticks,
+        parsedAmounts: derivedMintInfo.parsedAmounts,
+      };
+    }
+    return {
+      ...derivedMintInfo,
+    };
+  }, [derivedMintInfo, baseCurrency, quoteCurrency]);
+
+  const initialUSDPrices = useInitialUSDPrices();
+  const usdPriceA = useUSDCPrice(baseCurrency ?? undefined);
+  const usdPriceB = useUSDCPrice(quoteCurrency ?? undefined);
+
+  const {
+    onFieldAInput,
+    onFieldBInput,
+    onLeftRangeInput,
+    onRightRangeInput,
+    onStartPriceInput,
+  } = useV3MintActionHandlers(mintInfo.noLiquidity);
+
+  const { startPriceTypedValue } = useV3MintState();
+
+  const handleCurrencySelect = useCallback(
+    (
+      currencyNew: Currency,
+      currencyIdOther?: string,
+    ): (string | undefined)[] => {
+      const currencyIdNew = currencyId(currencyNew, chainId || 137);
+
+      let chainSymbol;
+
+      if (chainId === 137) {
+        chainSymbol = 'MATIC';
+      }
+
+      resetState();
+
+      if (currencyIdNew.toLowerCase() === currencyIdOther?.toLowerCase()) {
+        // not ideal, but for now clobber the other if the currency ids are equal
+        return [currencyIdNew, undefined];
+      } else {
+        // prevent weth + eth
+        const isETHOrWETHNew =
+          currencyIdNew === chainSymbol ||
+          (chainId !== undefined &&
+            currencyIdNew === WMATIC_EXTENDED[chainId]?.address);
+        const isETHOrWETHOther =
+          currencyIdOther !== undefined &&
+          (currencyIdOther === chainSymbol ||
+            (chainId !== undefined &&
+              currencyIdOther === WMATIC_EXTENDED[chainId]?.address));
+
+        if (isETHOrWETHNew && isETHOrWETHOther) {
+          return [currencyIdNew, undefined];
+        } else {
+          return [currencyIdNew, currencyIdOther];
+        }
+      }
+    },
+    [chainId],
+  );
+
+  const handleCurrencyASelect = useCallback(
+    (currencyANew: Currency) => {
+      const [idA] = handleCurrencySelect(currencyANew);
+      setCurrencyIdA(idA);
+    },
+    [handleCurrencySelect],
+  );
+
+  const handleCurrencyBSelect = useCallback(
+    (currencyBNew: Currency) => {
+      const [idB] = handleCurrencySelect(currencyBNew);
+      setCurrencyIdB(idB);
+    },
+    [handleCurrencySelect],
+  );
+
+  const handleCurrencySwap = useCallback(() => {
+    // history.push(`/add/${currencyIdB}/${currencyIdA}`);
+    resetState();
+  }, [handleCurrencySelect, currencyIdA, currencyIdB]);
+
+  const handlePopularPairSelection = useCallback((pair: [string, string]) => {
+    const WMATIC = '0x0d500b1d8e8ef31e21c99d1db9a6444d3adf1270';
```

Comment:
shoud not be hard coded use from consts

https://polygonscan.com/address/0x0d500b1d8e8ef31e21c99d1db9a6444d3adf1270